### PR TITLE
[IOTDB-3360] Fix precision loss bug due to float converted to double

### PR DESCRIPTION
**Pull request by Plutooooooo**
More details: https://issues.apache.org/jira/browse/IOTDB-3360

### CODE REVIEW

**Review by Alima777 — COMMENTED** [review]
Revert the change about checkstyle.xml?

**Comment by Plutooooooo** [comment]
> Revert the change about checkstyle.xml?


Spotless made the change automatically. Fixed.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java [hunk 1]
```diff
@@ -138,6 +139,8 @@
 /** Parse AST to Statement. */
 public class ASTVisitor extends IoTDBSqlParserBaseVisitor<Statement> {
 
+  private static final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java:142]:
```suggestion
  private static final IoTDBConfig CONFIG = IoTDBDescriptor.getInstance().getConfig();
```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java [hunk 2]
```diff
@@ -2067,6 +2070,14 @@ private Expression parseConstantOperand(ConstantContext constantContext) {
     }
   }
 
+  private Expression parseRealLiteral(String value) {
+    if (config.getFloatingStringInferType().equals(TSDataType.DOUBLE)) {
+      return new ConstantOperand(TSDataType.DOUBLE, value);
+    }
+    // 3.33 is float by default
+    return new ConstantOperand(TSDataType.FLOAT, value);
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java:2078]:
```suggestion
      return new ConstantOperand(config.getFloatingStringInferType().equals(TSDataType.DOUBLE) ? TSDataType.DOUBLE : TSDataType.FLOAT, value);
```

Comment (Plutooooooo) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java]: Fixed.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java (hunk 1 above)
Comment (Plutooooooo) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/parser/ASTVisitor.java]: Fixed.

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**